### PR TITLE
chore: migrate goreleaser config to v2

### LINKED ISSUE #1104: Latest release `1.5.0` not marked as latest
I noticed that v1.5.0 was not marked as the latest release. I assume it is related to #1075 because the latest Goreleaser action uses version 2 and the config has not been adjusted accordingly.

I am currently preparing a PR that should fix it for future releases.

### PR DESCRIPTION
**Pull request by morphy2k**
This PR updates the GoReleaser configuration to v2.

Fixes #1104

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -1,31 +1,53 @@
...
   - id: revive
     name_template: '{{ .ProjectName }}_{{ .Os }}_{{ .Arch }}{{ if .Arm }}v{{ .Arm }}{{ end }}'
+    format: tar.gz
+
 before:
   hooks:
-    - "go mod download"
-    - "go generate ./..."
+    - go mod download
+    - go generate ./...
```
Comment (alexandear) [inline .goreleaser.yml:13]: Can we revert style changes that are not related to the topic of the PR?

**Review by alexandear — CHANGES_REQUESTED** [review]

File: .goreleaser.yml (hunk 1 above)
Comment (alexandear) [reply .goreleaser.yml:13]: I propose to create two PRs:

1) the fix for https://github.com/mgechev/revive/issues/1104
2) the PR for updating GoReleaser configuration.

**Review by alexandear — APPROVED** [review]

**Comment by alexandear** [comment]
This PR fixes #1104?

**Comment by denisvmedia** [comment]
Thank you!

**Pull request merged**